Flaky test test_013_create_l2vpn_with_vlan_range

Last two executions of the daily pipeline resulted in the following errors:

```
___________ TestE2EReturnCodes.test_013_create_l2vpn_with_vlan_range ___________

self = <test_06_l2vpn_return_codes.TestE2EReturnCodes object at 0x7f378df89a10>

   def test_013_create_l2vpn_with_vlan_range(self):
       """
       Test the return code for creating a SDX L2VPN
       201: L2VPN Service Created
       P2P with VLAN range
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
       payload = {
           "name": "Test L2VPN creation with VLANs range",
           "endpoints": [
               {"port_id": "urn:sdx:port:[ampath.net](http://ampath.net/):Ampath3:50","vlan": "600:999"},
               {"port_id": "urn:sdx:port:[sax.net](http://sax.net/):Sax01:50","vlan": "600:999"}
           ]
       }
       response = requests.post(api_url, json=payload)
       assert response.status_code == 201, response.text
       data = response.json()
       assert data.get("status") == "under provisioning", data
       service_id = data.get("service_id")
       assert service_id != None, data

       # give enough time to SDX-Controller to propagate change to OXPs
       time.sleep(5)

       response = requests.get(api_url)
       assert response.status_code == 200, response.text
       data = response.json()
     assert len(data) == 1, data
E       AssertionError: {'0fda340b-b24a-4acd-9c28-ec06e86395b9': {'archived_date': 0, 'current_path': [{'port_id': 'urn:sdx:port:[ampath.net](http://ampath.net/):Am...:[ampath.net](http://ampath.net/):Ampath3:50', 'vlan': '311'}, {'port_id': 'urn:sdx:port:[tenet.ac.za](http://tenet.ac.za/):Tenet01:50', 'vlan': '311'}], ...}, ...}
E       assert 12 == 1
E        +  where 12 = len({'0fda340b-b24a-4acd-9c28-ec06e86395b9': {'archived_date': 0, 'current_path': [{'port_id': 'urn:sdx:port:[ampath.net](http://ampath.net/):Am...:[ampath.net](http://ampath.net/):Ampath3:50', 'vlan': '311'}, {'port_id': 'urn:sdx:port:[tenet.ac.za](http://tenet.ac.za/):Tenet01:50', 'vlan': '311'}], ...}, ...})

tests/test_06_l2vpn_return_codes.py:119: AssertionError
```

Although the issue happened on test test_013_create_l2vpn_with_vlan_range (tests/test_06_l2vpn_return_codes.py), the failed assertion indicates that the SDX-COntroller had 11 unexpected L2VPNs at that point. So it may be related to previous tests that propagated the error to this one.

--- a/tests/test_05_l2vpn.py
+++ b/tests/test_05_l2vpn.py
@@ -55,10 +55,10 @@ class TestE2EL2VPN:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
-        response_json = response.json()
-        assert response_json.get("status") == "under provisioning", response_json
-        service_id = response_json.get("service_id")
-        assert service_id != None, response_json
+        data = response.json()
+        assert data.get("status") == "under provisioning", str(data)
+        service_id = data.get("service_id")
+        assert service_id != None, str(data)
 
         # give enough time to SDX-Controller to propagate change to OXPs
         time.sleep(10)
@@ -66,10 +66,10 @@ class TestE2EL2VPN:
         api_url = SDX_CONTROLLER + '/l2vpn/1.0'
         response = requests.get(api_url)
         assert response.status_code == 200, response.text
-        response_json = response.json()
-        assert len(response_json) == 1, response_json
-        assert service_id in response_json, response_json
-        assert response_json[service_id].get("status") == "up", response_json
+        data = response.json()
+        assert len(data) == 1, str(data)
+        assert service_id in data, str(data)
+        assert data[service_id].get("status") == "up", str(data)
 
         # make sure OXPs have the new EVCs
         ## -> ampath
@@ -80,10 +80,11 @@ class TestE2EL2VPN:
         for evc in evcs.values():
             if evc.get("uni_a", {}).get("tag", {}).get("value") == 300:
                 found += 1
-        assert found == 1, evcs
+        assert found == 1, str(evcs)
         ## -> sax
         response = requests.get("http://sax:8181/api/kytos/mef_eline/v2/evc/")
-        assert len(response.json()) == 1, response.json()
+        data = response.json()
+        assert len(data) == 1, str(data)
         ## -> tenet
         response = requests.get("http://tenet:8181/api/kytos/mef_eline/v2/evc/")
         evcs = response.json()
@@ -92,7 +93,7 @@ class TestE2EL2VPN:
         for evc in evcs.values():
             if evc.get("uni_z", {}).get("tag", {}).get("value") == 300:
                 found += 1
-        assert found == 1, evcs
+        assert found == 1, str(evcs)
 
     def test_030_create_l2vpn_with_any_vlan(self):
         """Test creating a L2VPN successfully."""
@@ -112,10 +113,10 @@ class TestE2EL2VPN:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
-        response_json = response.json()
-        assert response_json.get("status") == "under provisioning", response_json
-        service_id = response_json.get("service_id")
-        assert service_id != None, response_json
+        data = response.json()
+        assert data.get("status") == "under provisioning", str(data)
+        service_id = data.get("service_id")
+        assert service_id != None, str(data)
 
         # give enough time to SDX-Controller to propagate change to OXPs
         time.sleep(10)
@@ -123,10 +124,10 @@ class TestE2EL2VPN:
         api_url = SDX_CONTROLLER + '/l2vpn/1.0'
         response = requests.get(api_url)
         assert response.status_code == 200, response.text
-        response_json = response.json()
-        assert len(response_json) == 2, response_json
-        assert service_id in response_json, response_json
-        assert response_json[service_id].get("status") == "up", response_json
+        data = response.json()
+        assert len(data) == 2, str(data)
+        assert service_id in data, str(data)
+        assert data[service_id].get("status") == "up", str(data)
 
         # make sure OXPs have the new EVCs
         ## -> ampath
@@ -149,7 +150,7 @@ class TestE2EL2VPN:
         for evc in evcs.values():
             if evc.get("uni_a", {}).get("tag", {}).get("value") == 100:
                 found += 1
-        assert found == 0, evcs
+        assert found == 0, str(evcs)
         ## -> tenet
         response = requests.get("http://tenet:8181/api/kytos/mef_eline/v2/evc/")
         evcs = response.json()
@@ -157,7 +158,7 @@ class TestE2EL2VPN:
         for evc in evcs.values():
             if evc.get("uni_z", {}).get("tag", {}).get("value") == 100:
                 found += 1
-        assert found == 0, evcs
+        assert found == 0, str(evcs)
 
         # wait a few seconds to allow status change from UNDER_PROVISIONG to UP
         time.sleep(5)
@@ -204,7 +205,7 @@ class TestE2EL2VPN:
         for evc in evcs.values():
             if evc.get("uni_a", {}).get("tag", {}).get("value") == 100:
                 found += 1
-        assert found == 1, evcs
+        assert found == 1, str(evcs)
         ## -> tenet
         response = requests.get("http://tenet:8181/api/kytos/mef_eline/v2/evc/")
         evcs = response.json()
@@ -212,7 +213,7 @@ class TestE2EL2VPN:
         for evc in evcs.values():
             if evc.get("uni_z", {}).get("tag", {}).get("value") == 100:
                 found += 1
-        assert found == 1, evcs
+        assert found == 1, str(evcs)
 
     def test_045_edit_port_l2vpn_successfully(self):
         """Test change the port_id of endpoints of an existing L2vpn connection."""
@@ -221,10 +222,8 @@ class TestE2EL2VPN:
         data = response.json()
         key = list(data.keys())[0]
         current_data = data[key]  
-        assert current_data["endpoints"][0]["port_id"] in ["urn:sdx:port:tenet.ac.za:Tenet03:50", \
-                                                           "urn:sdx:port:ampath.net:Ampath3:50"], str(data)
-        assert current_data["endpoints"][1]["port_id"] in ["urn:sdx:port:tenet.ac.za:Tenet03:50", \
-                                                           "urn:sdx:port:ampath.net:Ampath3:50"], str(data)
+        assert current_data["endpoints"][0]["port_id"] == "urn:sdx:port:ampath.net:Ampath3:50", str(data)
+        assert current_data["endpoints"][1]["port_id"] == "urn:sdx:port:tenet.ac.za:Tenet03:50", str(data)
 
         # Change port_id
         payload = {

--- a/tests/test_05_l2vpn.py
+++ b/tests/test_05_l2vpn.py
@@ -216,9 +216,6 @@ class TestE2EL2VPN:
 
     def test_045_edit_port_l2vpn_successfully(self):
         """Test change the port_id of endpoints of an existing L2vpn connection."""
-        # wait a few seconds to allow status change from UNDER_PROVISIONG to UP
-        time.sleep(5)
-
         api_url = SDX_CONTROLLER + '/l2vpn/1.0'
         response = requests.get(api_url)
         data = response.json()
@@ -228,17 +225,17 @@ class TestE2EL2VPN:
                                                            "urn:sdx:port:ampath.net:Ampath3:50"], str(data)
         assert current_data["endpoints"][1]["port_id"] in ["urn:sdx:port:tenet.ac.za:Tenet03:50", \
                                                            "urn:sdx:port:ampath.net:Ampath3:50"], str(data)
-            
+
         # Change port_id
         payload = {
             "name": "New port_id in endpoints",
             "endpoints": [
                 {
-                    "port_id": "urn:sdx:port:tenet.ac.za:Tenet01:41",
+                    "port_id": "urn:sdx:port:ampath.net:Ampath2:50",
                     "vlan": "100",
                 },
                 {
-                    "port_id": "urn:sdx:port:sax.net:Sax01:40",
+                    "port_id": "urn:sdx:port:sax.net:Sax02:50",
                     "vlan": "100",
                 }
             ]
@@ -246,11 +243,27 @@ class TestE2EL2VPN:
         response = requests.patch(f"{api_url}/{key}", json=payload)
         assert response.status_code == 201, response.text
 
-        response = requests.get(api_url + f'/{key}/archived')
-        data = response.json()
-        current_data = data[key] 
-        assert current_data["current_path"][0]["port_id"] == "urn:sdx:port:tenet.ac.za:Tenet01:41", str(data)
-        assert current_data["current_path"][-1]["port_id"] == "urn:sdx:port:sax.net:Sax01:40", str(data)
+        # give enough time to SDX-Controller to propagate change to OXPs
+        time.sleep(10)
+
+        response = requests.get(f"{api_url}/{key}")
+        data = response.json()[key]
+        assert data["status"] == "up", str(data)
+        assert len(data["endpoints"]) == 2, str(data)
+        assert data["endpoints"][0]["port_id"] == "urn:sdx:port:ampath.net:Ampath2:50", str(data)
+        assert data["endpoints"][1]["port_id"] == "urn:sdx:port:sax.net:Sax02:50", str(data)
+
+
+        # make sure OXPs have the new EVCs
+        ## -> ampath
+        response = requests.get("http://ampath:8181/api/kytos/mef_eline/v2/evc/")
+        assert len(response.json()) == 2, response.text
+        ## -> sax
+        response = requests.get("http://sax:8181/api/kytos/mef_eline/v2/evc/")
+        assert len(response.json()) == 2, response.text
+        ## -> tenet
+        response = requests.get("http://tenet:8181/api/kytos/mef_eline/v2/evc/")
+        assert len(response.json()) == 1, response.text
 
     def test_050_delete_l2vpn_successfully(self):
         """Test deleting all two L2VPNs successfully."""
@@ -262,7 +275,7 @@ class TestE2EL2VPN:
         # Delete all L2VPN
         for key in data:
             response = requests.delete(f"{api_url}/{key}")
-            assert response.status_code == 200, response.text
+            assert response.status_code == 200, f"{response.text=} previous_data={data}"
 
         # give enough time to SDX-Controller to propagate change to OXPs
         time.sleep(10)

--- a/tests/test_06_l2vpn_return_codes.py
+++ b/tests/test_06_l2vpn_return_codes.py
@@ -71,6 +71,39 @@ class TestE2EReturnCodes:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
+        service_id = response.json()["service_id"]
+
+        # allow time for SDX-Controller propagate changes
+        time.sleep(5)
+
+        response = requests.get(f"{api_url}/{service_id}")
+        assert response.status_code == 200, response.text
+        data = response.json()[service_id]
+        assert data["status"] == "up", str(data)
+        assert len(data["endpoints"]) == 2, str(data)
+        assert len(data["current_path"]) > 0, str(data)
+        service_id = response.json()["service_id"]
+
+        # allow time for SDX-Controller propagate changes
+        time.sleep(5)
+
+        response = requests.get(f"{api_url}/{service_id}")
+        assert response.status_code == 200, response.text
+        data = response.json()[service_id]
+        assert data["status"] == "up", str(data)
+        assert len(data["endpoints"]) == 2, str(data)
+        assert len(data["current_path"]) > 0, str(data)
+        service_id = response.json()["service_id"]
+
+        # allow time for SDX-Controller propagate changes
+        time.sleep(5)
+
+        response = requests.get(f"{api_url}/{service_id}")
+        assert response.status_code == 200, response.text
+        data = response.json()[service_id]
+        assert data["status"] == "up", str(data)
+        assert len(data["endpoints"]) == 2, str(data)
+        assert len(data["current_path"]) > 0, str(data)
 
     def test_012_create_l2vpn_with_vlan_any(self):
         """
@@ -534,6 +567,9 @@ class TestE2EReturnCodes:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
         payload = {
             "name": "Test L2VPN creation available bw",
             "endpoints": [
@@ -570,6 +606,9 @@ class TestE2EReturnCodes:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
         payload = {
             "name": "Test L2VPN creation no available bw",
             "endpoints": [
@@ -584,6 +623,18 @@ class TestE2EReturnCodes:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
+
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 2, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
 
     def test_055_create_l2vpn_with_valid_max_delay(self):
         """
@@ -604,6 +655,18 @@ class TestE2EReturnCodes:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
+
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 1, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
 
     def test_056_create_l2vpn_with_max_delay_out_of_range(self):
         """
@@ -667,6 +730,18 @@ class TestE2EReturnCodes:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 1, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
+
     def test_059_create_l2vpn_with_max_number_oxps_out_of_range(self):
         """
         Test the return code for creating a SDX L2VPN
@@ -728,6 +803,9 @@ class TestE2EReturnCodes:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
         payload = {
             "name": "Test L2VPN creation",
             "endpoints": [
@@ -764,6 +842,9 @@ class TestE2EReturnCodes:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
         payload = {
             "name": "Test L2VPN creation no available oxps",
             "endpoints": [
@@ -778,6 +859,18 @@ class TestE2EReturnCodes:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
+
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 2, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
 
     def test_070_create_l2vpn_with_impossible_scheduling(self):
         """

--- a/tests/test_06_l2vpn_return_codes.py
+++ b/tests/test_06_l2vpn_return_codes.py
@@ -65,7 +65,6 @@ class TestE2EReturnCodes:
         assert data["status"] == "up", str(data)
         assert len(data["endpoints"]) == 2, str(data)
         assert len(data["current_path"]) > 0, str(data)
-        service_id = response.json()["service_id"]
     
     def test_011_create_l2vpn_vlan_translation(self):
         """
@@ -94,7 +93,6 @@ class TestE2EReturnCodes:
         assert data["status"] == "up", str(data)
         assert len(data["endpoints"]) == 2, str(data)
         assert len(data["current_path"]) > 0, str(data)
-        service_id = response.json()["service_id"]
 
         # allow time for SDX-Controller propagate changes
         time.sleep(5)
@@ -105,7 +103,6 @@ class TestE2EReturnCodes:
         assert data["status"] == "up", str(data)
         assert len(data["endpoints"]) == 2, str(data)
         assert len(data["current_path"]) > 0, str(data)
-        service_id = response.json()["service_id"]
 
         # allow time for SDX-Controller propagate changes
         time.sleep(5)

--- a/tests/test_06_l2vpn_return_codes.py
+++ b/tests/test_06_l2vpn_return_codes.py
@@ -130,6 +130,17 @@ class TestE2EReturnCodes:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
+        service_id = response.json()["service_id"]
+
+        # allow time for SDX-Controller propagate changes
+        time.sleep(5)
+
+        response = requests.get(f"{api_url}/{service_id}")
+        assert response.status_code == 200, response.text
+        data = response.json()[service_id]
+        assert data["status"] == "up", str(data)
+        assert len(data["endpoints"]) == 2, str(data)
+        assert len(data["current_path"]) > 0, str(data)
     
     def test_013_create_l2vpn_with_vlan_range(self):
         """
@@ -482,7 +493,7 @@ class TestE2EReturnCodes:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
 
-        time.sleep(30)
+        time.sleep(5)
 
         response = requests.post(api_url, json=payload)
         assert response.status_code == 409, response.text
@@ -506,6 +517,17 @@ class TestE2EReturnCodes:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
+        service_id = response.json()["service_id"]
+
+        # give enough time to SDX-Controller to propagate change to OXPs
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 1, str(data)
+        assert service_id in data, str(data)
+        assert data[service_id].get("status") == "up", str(data)
 
     def test_051_create_l2vpn_with_min_bw_out_of_range(self):
         """

--- a/tests/test_06_l2vpn_return_codes.py
+++ b/tests/test_06_l2vpn_return_codes.py
@@ -54,6 +54,18 @@ class TestE2EReturnCodes:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
+        service_id = response.json()["service_id"]
+
+        # allow time for SDX-Controller propagate changes
+        time.sleep(5)
+
+        response = requests.get(f"{api_url}/{service_id}")
+        assert response.status_code == 200, response.text
+        data = response.json()[service_id]
+        assert data["status"] == "up", str(data)
+        assert len(data["endpoints"]) == 2, str(data)
+        assert len(data["current_path"]) > 0, str(data)
+        service_id = response.json()["service_id"]
     
     def test_011_create_l2vpn_vlan_translation(self):
         """

--- a/tests/test_06_l2vpn_return_codes.py
+++ b/tests/test_06_l2vpn_return_codes.py
@@ -148,9 +148,9 @@ class TestE2EReturnCodes:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
         data = response.json()
-        assert data.get("status") == "under provisioning", data
+        assert data.get("status") == "under provisioning", str(data)
         service_id = data.get("service_id")
-        assert service_id != None, data
+        assert service_id != None, str(data)
 
         # give enough time to SDX-Controller to propagate change to OXPs
         time.sleep(5)
@@ -158,9 +158,9 @@ class TestE2EReturnCodes:
         response = requests.get(api_url)
         assert response.status_code == 200, response.text
         data = response.json()
-        assert len(data) == 1, data
-        assert service_id in data, data
-        assert data[service_id].get("status") == "up", data
+        assert len(data) == 1, str(data)
+        assert service_id in data, str(data)
+        assert data[service_id].get("status") == "up", str(data)
 
         #
         # make sure OXPs have the new EVCs
@@ -172,7 +172,7 @@ class TestE2EReturnCodes:
         for evc in evcs.values():
             if evc.get("uni_a", {}).get("tag", {}).get("value") == [[600, 999]]:
                 found += 1
-        assert found == 1, evcs
+        assert found == 1, str(evcs)
         ## -> sax
         response = requests.get("http://sax:8181/api/kytos/mef_eline/v2/evc/")
         evcs = response.json()
@@ -181,7 +181,7 @@ class TestE2EReturnCodes:
         for evc in evcs.values():
             if evc.get("uni_z", {}).get("tag", {}).get("value") == [[600, 999]]:
                 found += 1
-        assert found == 1, evcs
+        assert found == 1, str(evcs)
 
     def test_014_create_l2vpn_with_vlan_untagged(self):
         """
@@ -200,9 +200,9 @@ class TestE2EReturnCodes:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
         data = response.json()
-        assert data.get("status") == "under provisioning", data
+        assert data.get("status") == "under provisioning", str(data)
         service_id = data.get("service_id")
-        assert service_id != None, data
+        assert service_id != None, str(data)
 
         # give enough time to SDX-Controller to propagate change to OXPs
         time.sleep(5)
@@ -210,9 +210,9 @@ class TestE2EReturnCodes:
         response = requests.get(api_url)
         assert response.status_code == 200, response.text
         data = response.json()
-        assert len(data) == 1, data
-        assert service_id in data, data
-        assert data[service_id].get("status") == "up", data
+        assert len(data) == 1, str(data)
+        assert service_id in data, str(data)
+        assert data[service_id].get("status") == "up", str(data)
 
         #
         # make sure OXPs have the new EVCs
@@ -224,7 +224,7 @@ class TestE2EReturnCodes:
         for evc in evcs.values():
             if evc.get("uni_a", {}).get("tag", {}).get("value") == 599:
                 found += 1
-        assert found == 1, evcs
+        assert found == 1, str(evcs)
         ## -> sax
         response = requests.get("http://sax:8181/api/kytos/mef_eline/v2/evc/")
         evcs = response.json()
@@ -237,7 +237,7 @@ class TestE2EReturnCodes:
         for evc in evcs.values():
             if evc.get("uni_z", {}).get("tag", {}).get("value") == "untagged":
                 found += 1
-        assert found == 1, evcs
+        assert found == 1, str(evcs)
 
     def _future_date(self, isoformat_time=True):
         # Get the current time

--- a/tests/test_07_l2vpn_return_codes.py
+++ b/tests/test_07_l2vpn_return_codes.py
@@ -70,6 +70,18 @@ class TestE2EReturnCodesEditL2vpn:
         response = requests.patch(f"{api_url}/{self.key}", json=self.payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 1, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
+
     def test_011_edit_l2vpn_port_id(self):
         """
         Test the return code for editing a SDX L2VPN
@@ -80,6 +92,18 @@ class TestE2EReturnCodesEditL2vpn:
         self.payload['endpoints'][0]['port_id'] = "urn:sdx:port:sax.net:Sax01:50"
         response = requests.patch(f"{api_url}/{self.key}", json=self.payload)
         assert response.status_code == 201, response.text
+
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 1, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
 
     def test_020_edit_l2vpn_with_vlan_integer(self):
         """
@@ -254,6 +278,18 @@ class TestE2EReturnCodesEditL2vpn:
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 2, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
+
         # Edit the first l2pvn to match the newly created one
         self.payload['endpoints'][0]['vlan'] = "500"
         self.payload['endpoints'][1]['vlan'] = "500"
@@ -281,6 +317,18 @@ class TestE2EReturnCodesEditL2vpn:
         response = requests.patch(f"{api_url}/{self.key}", json=payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 1, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
+
     def test_061_edit_l2vpn_with_max_delay(self):
         """
         Test the return code for editing a SDX L2VPN
@@ -302,6 +350,18 @@ class TestE2EReturnCodesEditL2vpn:
         response = requests.patch(f"{api_url}/{self.key}", json=payload)
         assert response.status_code == 201, response.text
 
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 1, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
+
     def test_062_edit_l2vpn_with_max_number_oxps(self):
         """
         Test the return code for editing a SDX L2VPN
@@ -322,6 +382,18 @@ class TestE2EReturnCodesEditL2vpn:
         }
         response = requests.patch(f"{api_url}/{self.key}", json=payload)
         assert response.status_code == 201, response.text
+
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 1, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
 
     def test_063_edit_l2vpn_with_min_bw_out_of_range(self):
         """
@@ -409,6 +481,18 @@ class TestE2EReturnCodesEditL2vpn:
         }
         response = requests.post(api_url, json=payload)
         assert response.status_code == 201, response.text
+
+        # allow time for SDX-Controller to propagate changes
+        time.sleep(5)
+
+        response = requests.get(api_url)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data) == 2, str(data)
+        for service_id, l2vpn in data.items():
+            assert l2vpn["status"] == "up", str(l2vpn)
+            assert len(l2vpn["endpoints"]) == 2, str(l2vpn)
+            assert len(l2vpn["current_path"]) > 0, str(l2vpn)
 
         # Edit high min_bw to get code 410
         payload = {

